src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -33,14 +37,31 @@ pub struct CommandRunner {
   underlying: Arc<dyn crate::CommandRunner>,
   metadata: ProcessMetadata,
   store: Store,
-  action_cache_client: Arc<bazel_protos::remote_execution_grpc::ActionCacheClient>,
+  action_cache_client: Arc<ActionCacheClient<Channel>>,
   headers: BTreeMap<String, String>,
   platform: Platform,
   cache_read: bool,
   cache_write: bool,
 }
 
 impl CommandRunner {
+  fn create_tonic_endpoint(
```

Comment:
This is the third place I've seen this code... Maybe extract it to somewhere shared?

Reply:
Refactored.